tests/end_to_end_tests/managed_packages_tests/test_packages.py

```diff
@@ -167,81 +189,400 @@ def test_packages(
+    """
+    def _add_config(config):
+        agent_paths.agent_config_path.write_text(json.dumps(config))
+
+    # 1. Configure invalid path for "ca_cert_path" and verify agent throws and fails to start
+    logger.info("Performing invalid ca_cert_path config option checks")
+    invalid_ca_cert_path_config = default_config.copy()
+    invalid_ca_cert_path_config["ca_cert_path"] = "/tmp/invalid/ca_certs.crt"
+    _add_config(invalid_ca_cert_path_config)
+
+    agent_commander.start()
+
+    with pytest.raises(subprocess.CalledProcessError) as err_info:
+        agent_commander.get_status()
+
+    assert b"The agent does not appear to be running." in err_info.value.stdout
+
+    agent_log = agent_paths.agent_log_path.read_text()
+    assert "ca_cert_path: /tmp/invalid/ca_certs.crt" in agent_log
+    assert "failedAgentMain" in agent_log
+    assert (
+        'Invalid path "/tmp/invalid/ca_certs.crt" specified for the "ca_cert_path"'
+        in agent_log
+    )
+    _clear_agent_dirs_and_print_log(agent_commander)
+
+    # 2. Configure agent to use system CA bundle to verify the server cert and verify it works
+    logger.info("Performing system CA bundle checks")
+    system_ca_bundle_config = default_config.copy()
+    system_ca_bundle_config["ca_cert_path"] = "/etc/ssl/system-ca-bundle.crt"
+    _add_config(system_ca_bundle_config)
+    ubuntu_certs = pl.Path("/etc/ssl/certs/ca-certificates.crt")
+    fedora_certs = pl.Path("/etc/ssl/certs/ca-bundle.crt")
+    if ubuntu_certs.is_file():
+        system_ca_bundle_path = ubuntu_certs
+    elif fedora_certs.is_file():
+        system_ca_bundle_path = fedora_certs
+    else:
+        raise Exception("No system CA bundle found")
+
+    # NOTE: We create a symlink so we can re-use config with a fixed value for ca_cert_path
+    logger.info(f"Using system ca bundle: ${system_ca_bundle_path}")
+
+    system_ca_bundle_path_symlink_path = pl.Path("/etc/ssl/system-ca-bundle.crt")
+    system_ca_bundle_path_symlink_path.symlink_to(system_ca_bundle_path)
+    agent_commander.start()
+
+    def _wait_for_string_in_log(text):
+        while text not in agent_paths.agent_log_path.read_text():
+            timeout_tracker.sleep(1, f"Can't wait for a text '{text}' in the agent log")
+
+    _wait_for_string_in_log("HttpConnection uses native os ssl")
+    agent_log = agent_paths.agent_log_path.read_text()
+    assert "ca_cert_path: /etc/ssl/system-ca-bundle.crt" in agent_log
+    assert "HttpConnection uses native os ssl" in agent_log
+
+    agent_status = agent_commander.get_status()
+    assert "Last successful communication with Scalyr:" in agent_status
+    assert "Last successful communication with Scalyr: Never" not in agent_status
+    assert "Last copy response status:                 success" in agent_status
+    _stop_agent_and_remove_logs_and_data(agent_commander=agent_commander)
+
+    # 3. Mimic MITM attack by pointing a random domain to agent.scalyr.com and verify hostname
```

Comment:
Nicely done!

Reply:
Thanks, but that's already existing checks and I just moved them here :sweat_smile: